apps/src/applab/firebaseStorage.js

```diff
@@ -398,4 +400,82 @@ FirebaseStorage.renameColumn = function (tableName, oldName, newName, onSuccess,
     .then(onSuccess, onError);
 };
 
+/**
+ * Parses a CSV string into records data in a format that can be written into the
+ * records node of a firebase table.
+ * @param {string} csvData
+ * @returns {Promise} A promise containing recordsData or an error message.
+ */
+function parseRecordsDataFromCsv(csvData) {
+  return new Promise((resolve, reject) => {
+    parseCsv(csvData, {columns: true}, (error, records) => {
+      if (error) {
+        return reject(error);
+      }
+      resolve(records);
+    });
+  }).then(records => {
+    let recordsData = {};
+    records.forEach((record, index) => {
+      const id = index + 1;
+      Object.keys(record).forEach(key => {
+        record[key] = castValue(record[key]);
+      });
```

Comment:
I've changed this to `for (const key in record)` since it's more concise and uses the standard language. I think ES6 is expressive enough in each of these cases that we don't really need lodash, but if there are specific instances that seem are hard to read without it, I'd be happy to use lodash in those instances.